assets/src/design-system/components/toggle/index.js

```diff
@@ -0,0 +1,191 @@
+const TOGGLE_WIDTH = 44;
+const CIRCLE_DIAMETER = 28;
+const CIRCLE_INITIAL_POSITION = (TOGGLE_HEIGHT - CIRCLE_DIAMETER) / 2;
+const CIRCLE_FINAL_POSITION =
+  TOGGLE_WIDTH - CIRCLE_DIAMETER - CIRCLE_INITIAL_POSITION;
+
+const ICON_WIDTH = 7;
+const ICON_TOP_POSITION = 8;
+const ICON_LEFT_POSITION = 30.5;
+
+const Background = styled.div(
+  ({ theme }) => css`
+    position: absolute;
+    top: -${BORDER_WIDTH}px;
+    left: -${BORDER_WIDTH}px;
+    height: ${TOGGLE_HEIGHT}px;
+    width: ${TOGGLE_WIDTH}px;
+    background-color: transparent;
+    border-radius: ${theme.borders.radius.x_large};
+    border: ${BORDER_WIDTH}px solid ${theme.colors.border.defaultNormal};
+    pointer-events: none;
+    transition: all 0.3s;
+  `
+);
+
+const Circle = styled.span(
+  ({ theme }) => css`
+    pointer-events: none;
+
+    :after {
+      content: '';
+      position: absolute;
+      top: ${CIRCLE_INITIAL_POSITION}px;
+      left: ${CIRCLE_INITIAL_POSITION}px;
+      height: ${CIRCLE_DIAMETER}px;
+      width: ${CIRCLE_DIAMETER}px;
+      background-color: ${theme.colors.fg.secondary};
+      border-radius: ${theme.borders.radius.round};
+      cursor: pointer;
+      transition: all 0.3s;
+    }
+  `
+);
+
+const StyledCheckmark = styled(Checkmark)(
+  ({ theme }) => css`
+    position: absolute;
+    width: ${ICON_WIDTH}px;
+    top: ${ICON_TOP_POSITION}px;
+    left: ${ICON_LEFT_POSITION}px;
+    z-index: ${THEME_CONSTANTS.Z_INDEX.DEFAULT};
+    opacity: 0;
+    transition: opacity 0.3s;
+    pointer-events: none;
+
+    path {
+      fill: ${theme.colors.standard.white};
+    }
+  `
+);
+
+const ToggleContainer = styled.div(
+  ({ theme }) => css`
+    position: relative;
+    height: ${TOGGLE_HEIGHT}px;
+    width: ${TOGGLE_WIDTH}px;
+    background-color: ${theme.colors.bg.primary};
+    border-radius: ${theme.borders.radius.x_large};
+
+    input[type='checkbox'] {
+      position: absolute;
+      top: -${BORDER_WIDTH / 2}px;
+      left: -${BORDER_WIDTH / 2}px;
+      height: ${TOGGLE_HEIGHT + BORDER_WIDTH}px;
+      width: ${TOGGLE_WIDTH + BORDER_WIDTH}px;
+      margin: 0;
+      opacity: 0;
+      cursor: pointer;
+
+      :disabled {
+        ~ ${Background} {
+          border-color: ${theme.colors.fg.disable};
+        }
+
+        :checked ~ ${Background} {
+          background-color: ${theme.colors.fg.disable};
+          border-color: ${theme.colors.fg.disable};
+        }
+
+        ~ ${Circle}:after {
+          background-color: ${theme.colors.fg.disable};
+        }
+
+        :checked ~ ${Circle}:after {
+          background-color: ${theme.colors.bg.secondary};
+        }
+
+        ~ ${StyledCheckmark} path {
+          fill: ${theme.colors.interactiveFg.disable};
+        }
+      }
+
+      &.${FOCUS_VISIBLE_SELECTOR} {
+        ~ ${Background} {
+          outline: none;
+          box-shadow: 0 0 0 5px ${theme.colors.bg.primary},
+            0 0 0 7px ${theme.colors.border.focus};
+        }
+      }
```

Comment:
we need wider focus styles here. 2px off of the border doesn't go around the button.